### PR TITLE
Feature 479043: Sorting on GET /forms endpoint

### PR DESCRIPTION
**Pull request by mokhld**
## Story Link
https://dev.azure.com/defragovuk/DEFRA-CDP/_workitems/edit/479043

## Description
This PR introduces sorting on the `GET /forms` endpoint. It allows sorting with `sortBy` with a value of either  `updatedAt` or `title`, both `asc` or `desc` and defaults to `updatedAt` with `desc` (most recent forms`.
In terms of the title sort, it takes into consideration the entire form title as outlined in the story.

Example meta response without sorting: 
```
{
    "data": [
    {
     ....forms
     },
    ...
    ],
    "meta": {
        "pagination": {
            "page": 1,
            "perPage": 24,
            "totalItems": 6,
            "totalPages": 1
        },
        "sorting": {
            "sortBy": "updatedAt",
            "order": "desc"
        }
    }
}
```

### CODE REVIEW

File: src/api/forms/repositories/form-metadata-repository.js [hunk 1]
```diff
@@ -28,30 +28,113 @@ export async function listAll() {
...
+      )
+
+    const skip = (page - 1) * perPage
+
+    const pipeline = []
+
+    /** @type {CollationOptions | null} */
+    let collation = null
+
+    switch (sortBy) {
```
Comment (alexluckett) [inline src/api/forms/repositories/form-metadata-repository.js:56]: This function is getting a bit long. Maybe pull the sort options out into a function?

e.g.
```
const sortOptions = getSortOptions(sortBy)
pipeline.push(sortOptions)
```

File: src/api/forms/repositories/form-metadata-repository.js [hunk 2]
```diff
@@ -28,30 +28,113 @@ export async function listAll() {
...
+      case 'updatedAt':
+        pipeline.push(
+          {
+            $addFields: {
+              updatedDateOnly: {
```
Comment (alexluckett) [inline src/api/forms/repositories/form-metadata-repository.js:61]: I'd pull these into a var so you can re-use them for the default pipeline. Saves them drifting in the future.

**Review by alexluckett — COMMENTED** [review]
Looking good!

File: src/api/forms/repositories/form-metadata-repository.js (hunk 1 above)
Comment (mokhld) [reply src/api/forms/repositories/form-metadata-repository.js]: Yeah good shout- that's been split up and into vars as well 👌

**Pull request merged**